arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+
+The request uses the following URI parameters.
+
+`Bucket`: Specifies the bucket being deleted.
+
+Required: Yes
+
+`x-amz-expected-bucket-owner`: The account id of the expected bucket owner. If the bucket is owned by a different account, the request will fail with an HTTP 403 (Access Denied) error.
+
+##### Request body:
+No body
+
+##### Response:
+return code 204
+
+#### Azure:
+Azure cloud bucket creation:
+
+##### Rules for Bucket(known as container in azure) naming:
+
+* A container name must be a valid DNS name, conforming to the following naming rules:
+
+* Container names must start with a letter or number, and can contain only letters, numbers, and the dash (-) character.
+
+* Every dash (-) character must be immediately preceded and followed by a letter or number; consecutive dashes are not permitted in container names.
+
+* All letters in a container name must be lowercase.
+* Container names must be from 3 through 63 characters long.
+
+##### API:
+
+***
+ ##### PUT https://myaccount.blob.core.windows.net/mycontainer?restype=container
+***
+
+##### Request headers:
+
+* Authorization     Required.
+
+* Date or x-ms-date     Required.  (Can be automated or manual)
+
+* x-ms-version     Required for all authorized requests.
+
+  x-ms-meta-name:value     Optional.
+
+  x-ms-blob-public-access     Optional.
+
+  x-ms-client-request-id     Optional.
+
+##### Response headers:
+
+
+`return code`: 200
+
+`ETag`
+
+`Last-Modified`
+
+`x-ms-request-id`
+
+`x-ms-version`
+
+`Date`
+
+`x-ms-client-request-id`
+
+***
+##### DELETE    https://myaccount.blob.core.windows.net/mycontainer?restype=container
+****
+Request headers:
+
+`Timeout`:  Optional
+
+Request headers:
+
+* `Authorization`: Required.
+
+* `Date or x-ms-date`: Required.
+
+* `x-ms-lease-id`: <ID> Required for version 2012-02-12 and newer if the container has an active lease.
+
+* `x-ms-version`:  Required for all authorized requests.
+
+* `x-ms-client-request-id`:   Optional.
+
+Response:
+
+The response includes an HTTP status code and a set of response headers.
+
+`Status Code`: A successful operation returns status code 202 (Accepted).
+
+Response headers:
+
+None.
+
+
+#### GCP
+
+For GCP bucket creation:
+
+##### Rules for Bucket naming:
+
+ * Bucket names must contain only lowercase letters, numbers, dashes (-), underscores (*_)*, and dots (.). Spaces are not allowed
+
+ * Names containing dots require verification.
+
+* Bucket names must start and end with a number or letter.
+
+* Bucket names must contain 3-63 characters. Names containing dots can contain up to 222 characters, but each dot-separated component can be no longer than 63 characters.
+* Bucket names cannot be represented as an IP address in dotted-decimal notation (for example, 192.168.5.4).
+
+* Bucket names cannot begin with the "goog" prefix.
+
+* Bucket names cannot contain "google" or close misspellings, such as "g00gle".
+
+
+##### API:
+***
+POST https://storage.googleapis.com/storage/v1/b
```

Comment:
this is the API exposed by SODA ? if not, why do we expose a Google API outside ?

Reply:
This is not exposed by SODA. This doc has this API details for just reference and understanding.